src/tools/r2rdump/R2RMethod.cs

```diff
@@ -164,8 +164,15 @@ public R2RMethod(byte[] image, MetadataReader mdReader, uint rid)
             BlobReader signatureReader = mdReader.GetBlobReader(methodDef.Signature);
 
             DeclaringType = new Stack<string>();
-            TypeDefinition declaringTypeDef = mdReader.GetTypeDefinition(methodDef.GetDeclaringType());
-            DeclaringType.Push(mdReader.GetString(declaringTypeDef.Name));
+            TypeDefinitionHandle declaringTypeHandle = methodDef.GetDeclaringType();
+            TypeDefinition declaringTypeDef;
+            while (!declaringTypeHandle.IsNil)
```

Comment:
The looping mechanism looks good. If you are only using the `DeclaringType` Stack property for the purposes of printing out the fully qualified name, perhaps you can replace that property with a string property and in this loop you build up the fully qualified name by prepending each GetString call.

Reply:
This comment can be applied to the namespace lookup too. It looks like you're only using the Stack in the string printout, so you can just build up that string here instead of storing a Stack that will only be used for one specific purpose later.